compiler/rustc_arena/src/lib.rs

```diff
@@ -73,24 +67,26 @@ impl<T> TypedArenaChunk<T> {
                 ptr::drop_in_place(start);
                 start = start.offset(1);
             }
+            // To make Vec's drop impl not drop the contents
+            self.inner.set_len(0);
```

Comment:
shouldn't this entire code just be `self.inner.truncate(0)` then? Also then `len` argument to this function... is it equal to `self.inner.len()`? If not why?

Reply:
> shouldn't this entire code just be self.inner.truncate(0) then? Also then len argument to this function... is it equal to self.inner.len()? If not why?

The len argument is not equal to self.inner.len(). Thus it's not equivalent to truncate. The elements up to `count` are initialized, the ones beyond until self.inner.len() aren't. Calling truncate would drop uninitialized data. But this reminds me, writing this felt a bit off and now I get where the problem was: it needs to be `Vec<MaybeUnint<T>>` instead of `Vec<T>`.